main/src/kotlinx/team/infra/InfraPlugin.kt

```diff
@@ -29,6 +29,9 @@ class InfraPlugin : Plugin<Project> {
         configureTeamCityLogging()
         configureTeamCityConfigGenerator(extension.teamcity, extension.publishing)
 
+        extension.afterTeamCity {
```

Comment:
This makes it dependent on whether teamcity extension has been accessed. It doesn't contain absolutely required options, so it may be the case that `teamcity.overrideBuildNumber` is true (by default), but the override logic is not enabled.

Reply:
Now `afterEvaluate {}` is used.